### ISSUE TITLE
Protractor synchronization does not work

### ISSUE THREAD

**Issue by Ijee** [jan-molak/serenity-js#343]
Hey, 
I have a problem on an updated version of a website that I wrote tests for with serenity-js. Namely the protractor synchronization does not work at all and it just waits for the global timeout with no interactions being run at all.

Is there any way I can debug this problem as it is quite annoying to check for all explicit waits I would have to add.

I know this is more of an protractor specific problem but I thought maybe someone encountered this one before. 
Edit: Yes, the website is still running angular with v7.2.15

**Comment by jan-molak** [comment]
Hey @Ijee - it will be difficult to suggest anything without being able to see the issues you're seeing, I'm afraid. As you said, all the synchronisation with Angular is done by Protractor, and Serenity/JS is not really involved in the process.

Having said that, you mention that the website you're testing has been updated? It might be a bit of a long shot, but one thing I noticed with new versions of Angular is that when people use Zone.js incorrectly (i.e. mix up `setInterval` and `setTimeout` with zone.js) it confuses Protractor.

Check out [this article](https://medium.com/@magnattic/preventing-the-infamous-protractor-timeouts-the-smart-way-66b8cc517b04) and let me know if that helped at all?

**Comment by Ijee** [comment]
I am sorry. I completely forgot to answer here.

I took a look at the source code of the site in question but I could not find any problems that had been mentioned in the article you linked and kind of gave up with it. I also noticed that explicit waits like Wait.until(<Target>, Is.Clickable()) fails after a very short amount of time (probably less than a second even though it should wait for a maximum amount of 5 seconds).

One of the buttons on the site is just an anchor tag that has `disabled` attribute based on the forms that I am filling out. I'd expect Is.Clickable() to work on an element like that.

I am really not sure what's wrong with the project and it really breaks the tests I wrote at the moment. You can't really trust the report when tests randomly fail.

**Comment by Ijee** [comment]
I finally came around to take another look at my problem and found [this](https://stackoverflow.com/questions/48627074/how-to-track-which-async-tasks-protractor-is-waiting-on/57854350#57854350)
answer from stackoverflow. 

It is not the best solution to tackle this issue but it does work and I got to find what blocked the protractor synchronization. Maybe it is useful for someone in the future until there is an easier way to debug issues like these with protractor.
